pkg/configv2/types/bacalhau.go

```diff
@@ -0,0 +1,118 @@
+package types
+
+type Bacalhau struct {
+	API     API    `yaml:"API,omitempty"`
+	DataDir string `yaml:"DataDir,omitempty"`
+
+	// TODO Should NameProvider be a cli flag only option since it is not used post
+	//   node creation?
+	NameProvider string `yaml:"NameProvider,omitempty"`
+
+	StrictVersionMatch  bool                `yaml:"StrictVersionMatch,omitempty"`
+	Orchestrator        Orchestrator        `yaml:"Orchestrator,omitempty"`
+	Compute             Compute             `yaml:"Compute,omitempty"`
+	WebUI               WebUI               `yaml:"WebUI,omitempty"`
+	InputSources        InputSourcesConfig  `yaml:"InputSources,omitempty"`
+	Publishers          PublishersConfig    `yaml:"Publishers,omitempty"`
+	Executors           ExecutorsConfig     `yaml:"Executors,omitempty"`
+	ResultDownloaders   ResultDownloaders   `yaml:"ResultDownloaders,omitempty"`
+	JobDefaults         JobDefaults         `yaml:"JobDefaults,omitempty"`
+	JobAdmissionControl JobAdmissionControl `yaml:"JobAdmissionControl,omitempty"`
+	Logging             Logging             `yaml:"Logging,omitempty"`
+	UpdateConfig        UpdateConfig        `yaml:"UpdateConfig,omitempty"`
+	FeatureFlags        FeatureFlags        `yaml:"FeatureFlags,omitempty"`
+}
+
+type UpdateConfig struct {
+	Interval Duration `yaml:"Interval,omitempty"`
+}
+
+type FeatureFlags struct {
+	ExecTranslation bool `yaml:"ExecTranslation,omitempty"`
+}
+
+type API struct {
+	Address string     `yaml:"Address,omitempty"`
+	TLS     TLS        `yaml:"TLS,omitempty"`
+	Auth    AuthConfig `yaml:"Auth,omitempty"`
+}
+
+type TLS struct {
+	CertFile string `yaml:"Certificate,omitempty"`
+	KeyFile  string `yaml:"Key,omitempty"`
+	CAFile   string `yaml:"CAFile,omitempty"`
+}
+
+type WebUI struct {
+	Enabled bool   `yaml:"Enabled,omitempty"`
+	Listen  string `yaml:"Listen,omitempty"`
+}
+
+type InputSourcesConfig struct {
+	Disabled []string `yaml:"Disabled,omitempty"`
+
+	ReadTimeout   Duration            `yaml:"ReadTimeout,omitempty"`
+	MazSize       string              `yaml:"MazSize,omitempty"`
+	Decompression DecompressionConfig `yaml:"Decompression"`
+
+	Config map[string]map[string]interface{} `yaml:"Config,omitempty"`
+}
+
+type PublishersConfig struct {
+	Disabled []string                          `yaml:"Disabled,omitempty"`
+	Config   map[string]map[string]interface{} `yaml:"Config,omitempty"`
+}
+
+type ExecutorsConfig struct {
+	Disabled []string                          `yaml:"Disabled,omitempty"`
+	Config   map[string]map[string]interface{} `yaml:"Config,omitempty"`
```

Comment:
Yeah correct

Reply:
It will look similar to this:
```yaml
Disabled:
  - executor1
  - executor2
Config:
  executor3:
    key1: value1
    key2: 42
  executor4:
    setting: "some setting"
    happy: true
    timeout: 30
 ```
A few things to consider with this pattern:
1. It will require a comprehensive list of available executors (and providers in general) for users to reference in order to determine which they want to disable.
2. It will be challenging to make work with the `config set` command.
3. We'll need to implement manual validation for each type, meaning we need a "schema" (as go code for example)
4. migrations may also be challenging.
5. It lacks type safety. 

I lean more towards a strongly typed design, and later, we can introduce this dynamic type when there is a strong need - right now I can't think of a strong reason supporting this design - aside from plugable executors (which we don't really support right now). But happy to do as you prefer.